File: Algorithms/p66_Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> res(digits.rbegin(),digits.rend());
        int i = 0;
        for(i=0;i<res.size();++i){
            ++res[i];
            if(res[i]==10) res[i] = 0;
            else break;
        }
        if(i==res.size()) res.push_back(1);
        reverse(res.begin(),res.end());
        return res;
    }
};

File: Algorithms/p66_Plus_One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        res = list(reversed(digits))
        carry = True
        for i in range(0,len(res)):
            res[i] += 1
            if res[i]==10:
                res[i] = 0
            else:
                carry = False
                break
        if carry:
            res.append(1)
        res.reverse()
        return res
